src/records/RecordsConfig.cc

```diff
@@ -805,7 +805,7 @@ static const RecordElement RecordsConfig[] =
   ,
   //  # default the ram cache size to AUTO_SIZE (-1)
   //  # alternatively: 20971520 (20MB)
-  {RECT_CONFIG, "proxy.config.cache.ram_cache.size", RECD_INT, "-1", RECU_RESTART_TS, RR_NULL, RECC_STR, "^-?[0-9]+$", RECA_NULL}
+  {RECT_CONFIG, "proxy.config.cache.ram_cache.size", RECD_INT, "-1", RECU_RESTART_TS, RR_NULL, RECC_STR, "^-?[0-9]+[A-Za-z]{0,}$", RECA_NULL}
```

Comment:
where does it evaluate the unit suffix to convert it into the appropriate int? does it check for invalid units? Seems like in the docs `proxy.config.cache.ram_cache_cutoff` also supports unit suffix. Should that regex be included as well?

Reply:
Thanks for having a look at this!

>where does it evaluate the unit suffix to convert it into the appropriate int? does it check for invalid units?

It should be handled [here](https://github.com/apache/trafficserver/blob/4681638c043a23834cec85a2d55a0cddd071881f/src/tscore/ParseRules.cc#L51). 
> Seems like in the docs proxy.config.cache.ram_cache_cutoff also supports unit suffix. Should that regex be included as well?

yes, and probably there are others as well. This particular line was changed because it was detected at `records.yaml` loading time.

Worse case scenario, it will be detected in the `ink_atoi`.  I can probably put another PR and make sure the regex are in place for records like this.